### DISCUSSION TITLE
Steay Calculation in THM

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#25117] [Q&A General]
@joshuahansel  Hi Josh, I found the error in my code and I've fixed it. Now I'm testing the affect of some coefficients in the equations. I need the steady results but it costs so much time because I can only get the steady results by transient calculation(the benchmark has quite a number of channels).
So can you give any advice about how to change current `FlowChannel1Phase` into a steady version?

Ethan

**Comment by GiudGiud** [comment]
In general, you can:
- make a Steady solve from a Transient one by removing time derivative kernels
- make a Steady solve from a Transient one using a very very large dt (at the denominator in time derivatives)

For THM in particular I m not sure
Does THM not let you simply select a Steady executioner?

**Reply by Ethan-xj** [reply]
Thanks Giud.

> make a Steady solve from a Transient one using a very very large dt (at the denominator in time derivatives)

For this strategy, the subchannel code can't solve the case whose dt is very large now.

> make a Steady solve from a Transient one by removing time derivative kernels

And for this one, I've tried to comment out the time derivative kernel in `FlowModelSinglePhase`. But the calculation does not converge when I try a single pipe steady case. Should I comment out the code which is related to the initial conditions?

> Does THM not let you simply select a Steady executioner?

I think not. When I select the steady executioner, the error info is:
```
*** ERROR ***
The following error occurred in the object "THM:problem", of type "THMProblem".

Vector tag 'TIME' does not exist.

This may occur if you have a TimeKernel in your problem but did not specify a transient executioner.

```

**Reply by GiudGiud** [reply]
Can you try initializing very close to the solution?
Steady problem are harder to solve than Transient problems, initialization will help

**Reply by joshuahansel** [reply]
As for your error message, are you using an up-to-date version of THM? Steady support for FlowChannel1Phase was added roughly two months ago. It does require a decent initial guess to be successful (for example, don't guess the velocity to be zero). We have not exercised it much yet.

**Reply by joshuahansel** [reply]
Note that there are a lot of components that still do not support Steady. These should be able to be extended by optionally adding time kernels, as was done for `FlowChannel1Phase`.

**Reply by Ethan-xj** [reply]
@GiudGiud @joshuahansel I didn't check the update of THM recently. I will have a try in the up-to-date version of THM firstly.
Thank you all :)
